ros1_rosbag_storage_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+cmake_minimum_required(VERSION 3.5)
+project(ros1_rosbag_storage_vendor)
+
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+
+# We don't need to build on Windows for now
+if(WIN32)
+  return()
+endif()
+
+set(git_apply git apply)
+
+include(ExternalProject)
+# We have to include a number of patches to the rosbag
+# 1. rosbag1_encryption_patch includes https://github.com/ros/ros_comm/pull/1499, without it, pluginlib won't link
+# 2. rosbag1_for_ros2_patch patches the pluginlib usage:
+#    Missing boost/bind include
+#    Using pluginlib with boost results in compile errors on some systems, so we need to switch to pluginlib without boost
+# 3. rosbag1_ament_patch.diff patches the catkin package to ament and renames it
+#    Patching to ament is necessary as otherwise colcon needs to build a catkin package together with an ament package, which is impossible
+#    Renaming the package is necessary on some systems to not confuse the linker
+ExternalProject_Add(ros1_rosbag_storage
+  DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}/download
+  URL https://github.com/ros/ros_comm/archive/669fbd32d2f92cc295f4b024fcb2f982fddec0f0.zip
+  URL_MD5 4c8b4c33165b223870f5d77bb697bef6
+  TIMEOUT 60
+  SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/sources
+  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/rosbag_install
+  PATCH_COMMAND
+    ${git_apply} ${CMAKE_CURRENT_SOURCE_DIR}/rosbag1_encryption_patch.diff &&
```

Comment:
maybe we can put these patch files in to a `resource` subfolder?

Reply:
Done